Marlin/Marlin_main.cpp

```diff
@@ -499,27 +497,28 @@ static bool drain_queued_commands_P() {
  */
 void enqueuecommands_P(const char* pgcode) {
   queued_commands_P = pgcode;
```

Comment:
Makes sense. I'm just trying to avoid the extra work of flowcharting, but at some point I really should, so I better understand how the command queue works. When it comes down to it, maybe a second ring buffer is the best solution. I'm sure that we have enough leftover SRAM to be able to implement it. But yeah, that is probably "overkill" when you only have two potential stream sources.